src/Microsoft.Azure.WebJobs.Host/Hosting/WebJobsServiceCollectionExtensions.cs

```diff
@@ -89,6 +89,9 @@ public static IWebJobsBuilder AddWebJobs(this IServiceCollection services, Actio
             services.TryAddSingleton<IDistributedLockManager, InMemoryDistributedLockManager>();
             services.TryAddSingleton<IScaleMonitorManager, ScaleMonitorManager>();
             services.TryAddSingleton<ITargetScalerManager, TargetScalerManager>();
+            services.TryAddSingleton<IScaleManager, ScaleManager>();
+            services.AddHostedService<ScaleMonitorService>();
+            services.AddSingleton<IScaleMetricsRepository, NullScaleMetricsRepository>();
```

Comment:
What is the purpose of adding a null scale metric repository here over an `InMemoryScaleMetricsRepository`?

Reply:
we `NullScaleMetricsRepository` to resolve `IScaleMetricsRepository` in `ScaleMonitorService` as it's added by default in webjobs host.
`IScaleMetricsRepository` is implemented in function host